explainaboard/visualizers/draw_charts.py

```diff
@@ -0,0 +1,265 @@
+                ax.text(
+                    x=j,
+                    y=i,
+                    s=conf_matrix[i, j],
+                    va='center',
+                    ha='center',
+                    size='xx-large',
+                )
+        plt.ylabel(feature_names[0], fontsize=16)
+        ax.set_yticks(np.arange(len(sorted_names[0])))
+        ax.set_yticklabels(sorted_names[0])
+        plt.xlabel(feature_names[1], fontsize=16)
+        ax.set_xticks(np.arange(len(sorted_names[1])))
+        ax.set_xticklabels(sorted_names[1])
+        plt.title(f'confusion for {sys_name}', fontsize=18)
+        out_file = os.path.join(
+            output_dir, f'{feature_names[0]}_{feature_names[1]}_combo_{combo_idx}.png'
+        )
+        plt.savefig(out_file, format='png', bbox_inches='tight')
+
+
+def plot_buckets(
+    bucket_results: list[BucketAnalysisResult], output_dir: str, sys_names: list[str]
+) -> None:
+    """
+    Plot bucket results in a bar chart
+    :param bucket_results: The analyses to write out
+    :param output_dir: The directory to write to
+    :param sys_names: The names of the systems
+    """
+    feature_name = bucket_results[0].name
+
+    bucket0_intervals = [
+        x.bucket_interval for x in bucket_results[0].bucket_performances
+    ]
+    bucket0_names = [
+        x.metric_name for x in bucket_results[0].bucket_performances[0].performances
+    ]
+    for metric_id, metric_name in enumerate(bucket0_names):
+
+        performances: list[list[Performance]] = [
+            [x.performances[metric_id] for x in y.bucket_performances]
+            for y in bucket_results
+        ]
+        ys = [[x.value for x in y] for y in performances]
+
+        y_errs = None
+        if performances[0][0].confidence_score_low is not None:
+            y_errs = [
+                (
+                    [x.value - unwrap(x.confidence_score_low) for x in y],
+                    [unwrap(x.confidence_score_high) - x.value for x in y],
+                )
+                for y in performances
+            ]
+
+        make_bar_chart(
+            ys,
+            output_dir,
+            f'{feature_name}_{metric_name}',
+            output_fig_format='png',
+            fig_size=(8, 6),
+            sys_names=sys_names,
+            errs=y_errs,
+            title=None,
+            xlabel=feature_name,
+            xticklabels=bucket0_intervals,
+            ylabel=metric_name,
+        )
+
+
+def draw_charts_from_reports(
+    reports: list[str], output_dir: str, sys_names: list[str] | None = None
+) -> None:
+    """
+    Draw bar charts from report file generated from ExplainaBoard
+    :param reports: Reports to plot
+    :param output_dir:
+    :return:
+    """
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    # TODO(gneubig): This should get the system name from inside the report
+    if sys_names is None:
+        sys_names = [os.path.basename(x).replace('.json', '') for x in reports]
```

Comment:
Use `splitext` rather than ad-hoc code.
```suggestion
        sys_names = [os.path.splitext(os.path.basename(x))[0] for x in reports]
```